aopy/preproc/quality.py

```diff
@@ -1,14 +1,46 @@
 import copy
 import matplotlib
-from .. import precondition
-from .. import analysis
-from ..utils import print_progress_bar
-from ..visualization.base import plot_image_by_time
-
 import numpy as np
 import numpy.linalg as npla
 import scipy.signal as sps
 import matplotlib.pyplot as plt
+import traceback
+from tqdm.auto import tqdm
+
+from .. import precondition
+from .. import analysis
+from ..utils import print_progress_bar
+from ..visualization.base import plot_image_by_time
+from ..data.base import load_preproc_exp_data
+
+def detect_buggy_entries(preproc_dir, subjects, ids, dates):
+    bad_entries = []
+    entries = list(zip(subjects, dates, ids))
+
+    for subject, date, te in tqdm(entries): 
+        # Load data from bmi3d hdf 
+        try:
+            exp_data, exp_metadata = load_preproc_exp_data(preproc_dir, subject, te, date)
+        except:
+            print(f"Entry {subject} {date} {te} could not be loaded.")
+            traceback.print_exc()
+            bad_entries.append([subject,date,te])
+            continue
+        
+        # Check events and times
+        try:
+            event_times = exp_data['events']['timestamp']
+        except:
+            print(f"Entry {subject} {date} {te} is missing event timestamps (likely missing ecube data).")
+            print('source files:', exp_metadata['source_files'])
+            bad_entries.append([subject,date,te])
+            continue
+        
+        if len(exp_data['events']) != len(exp_data['bmi3d_events']):
```

Comment:
can this be made even stronger by comparing the contents of `exp_data['events']['code']` to `exp_data['bmi3d_events']['code']` with something like np.array_equal?